src/analysis/Environments/DeleteButton.ts

```diff
@@ -18,19 +21,28 @@ export interface DeleteButtonProps {
 export const DeleteButton = (props: DeleteButtonProps): ReactNode => {
   const { resource, onClick } = props;
   const resourceType = isApp(resource) ? 'app' : 'runtime';
-  const isDeletable = isResourceDeletable(resourceType, resource);
+  // Casting resource to type App to access the appType property
+  const castToApp = resource as App;
```

Comment:
Ohh... sorry I'm a bit scattered today. I think I mistook the permissions provider for what this is doing. The permissions provider is telling you whether you `can` do something in terms of IAM. `isResourceDeletable` is telling you whether the resource is in a state that can be deleted. We should  not need to cast here though.

It sounds like we want to function `canBeDeleted` to take `resource: DecoratedComputeResource` and check  `isResourceDeletable` and the logic you have in `!Object.keys(cromwellAppToolLabels).includes(resource.appType);`. It will need to do something like the following
```
if (isApp(resource)) { // access appType here } 
else { //unimplemented exception,  other resources, aka runtimes, not implemented yet }
```

Bonus points if you can consolidate it with the `canDeleteDisk` function but you'd need to re-wire the type here from `DecoratedComputeResource` to `Resource`. 

Sorry if this is a bit confusing, definitely down to pair and help you get this through. You've got all the pieces in place here though, a great second iteration! tl;dr, we should not need to cast in the DeleteButton, the permissions provider should handle the guts of `isDeletable` and can take the resource directly and figure it out from there.

Reply:
I think I see what you're saying. Let me take a stab at this implementation, and I'll reach out if I get stuck. Thank you for all of the feedback!

Reply:
Also, just a quick clarification, are you advocating for calling `isResourceDeletable` from within `canBeDeleted`?

Reply:
I left that intentionally vague 😅 . I think the `permissionsProvider` ultimately was designed to do one thing only, tell you if you have permissions to delete something (which relates to IAM logic). 

I think we are encountering the need for another provider, one that can tell you whether a resource is in a state (business logic or otherwise) that allows a specific action. 

In the scope of your PR, I could see you doing any of the following, as long as we don't straddle between multiple approaches:
- extending the `permissionsProvider` with this `isResourceDeletable` functionality
- extending the utility function `isResourceDeletable` to include your cromwell-specific logic 
- creating a new provider that the delete button consumes that wraps the permissionsProvider and includes the utility function (isResourceDeletable) logic + your new cromwell app logic

Reply:
Sorry if I'm a bit scattered here, but trying to move towards an ideal architecture with old code through incremental improvements is a bit tricky 😅 . I'd vote for the third above, but I understand if you would rather go with something along the lines of #2 to not expand the scope of your PR too much

Reply:
No worries! I don't think it'll be too much of a lift to create a new wrapper provider, so I'll do my best to try and implement option #3 here. I'll reach out in Slack if things start to get a little wonky in my understanding.